docs/edgeql/literals.rst

```diff
@@ -0,0 +1,583 @@
+.. _ref_eql_literals:
+
+Literals
+========
+
+EdgeQL is *inextricably tied* to EdgeDB's rigorous type system. Below is an
+overview of how to declare a literal value of each *primitive type*. Click a
+link in the left column to jump to the associated section.
+
+.. list-table::
+
+  * - :ref:`String <ref_eql_literal_strings>`
+    - ``str``
+
+  * - :ref:`Boolean <ref_eql_literal_boolean>`
+    - ``bool``
+
+  * - :ref:`Numbers <ref_eql_literal_numbers>`
+    - ``int16`` ``int32`` ``int64``
+      ``float32`` ``float64`` ``bigint``
+      ``decimal``
+
+  * - :ref:`JSON <ref_eql_literal_json>`
+    - ``json``
+
+  * - :ref:`UUID <ref_eql_literal_uuid>`
+    - ``uuid``
+
+  * - :ref:`Enums <ref_eql_literal_enum>`
+    - ``enum<X, Y, Z>``
+
+  * - :ref:`Dates and times <ref_eql_literal_dates>`
+    - ``datetime`` ``duration``
+      ``cal::local_datetime`` ``cal::local_date``
+      ``cal::local_time`` ``cal::relative_duration``
+
+  * - :ref:`Durations <ref_eql_literal_durations>`
+    - ``duration`` ``cal::relative_duration``
+
+  * - :ref:`Bytes <ref_eql_literal_bytes>`
+    - ``bytes``
+
+  * - :ref:`Arrays <ref_eql_literal_array>`
+    - ``array<x>``
+
+  * - :ref:`Tuples <ref_eql_literal_tuple>`
+    - ``tuple<x, y, ...>`` or
+      ``tuple<foo: x, bar: y, ...>``
+
+
+.. _ref_eql_literal_strings:
+
+Strings
+-------
+
+The :eql:type:`str` type is a variable-length string of Unicode characters. A
+string can be declared with either single or double quotes.
+
+.. code-block:: edgeql
+
+  select 'i ❤️ edgedb';
+  select 'hello there!';
+  select r'hello\nthere!'; # raw string
+
+
+.. code-block:: edgeql-repl
+
+  db> select 'i ❤️ edgedb';
+  {'i ❤️ edgedb'}
+  db> select 'hello there!';
+  {'hello there!'}
+  db> select r'hello\nthere!';
+  {'hello
+  there!'}
+
+There is a special syntax for declaring "raw strings". Raw strings treat the
+backslash ``\`` as a literal character instead of an escape character.
+
+.. code-block:: edgeql-repl
+
+  db> select r'hello\nthere'; # raw string
+  {r'hello\\nthere'}
+  db> select $$one
+  ... two
+  ... three$$; # multiline raw string
```

Comment:
Actually all strings in EdgeQL can be multiline:

```
SELECT ' aaa

b

c'
```

is valid. Perhaps we should mention that. `$$` is cool for escaping strings, you can spell it in multiple ways: `$$`, `$my_string$` etc. Mention that briefly?

Reply:
What's the point of `$my_string$`?